### ISSUE TITLE
[Bug]: Api endpoint /api/video/{videoId}/sponsor/ returns a 500 when video doesn't exist

### ISSUE THREAD

**Issue by zac-schutt** [tubearchivist/tubearchivist#526]
### I've read the documentation

- [X] I'm running the latest version of Tube Archivist and have read the [release notes](https://github.com/tubearchivist/tubearchivist/releases/latest).
- [X] I have read the [how to open an issue](https://github.com/tubearchivist/tubearchivist/blob/master/CONTRIBUTING.md#how-to-open-an-issue) guide, particularly the [bug report](https://github.com/tubearchivist/tubearchivist/blob/master/CONTRIBUTING.md#bug-report) section.

### Operating System

Docker

## Describe the bug

When requesting sponsor segments for a video that is not in the library, an HTTP error 500 is returned.

## Steps To Reproduce

Send authenticated GET request to `/api/video/E8LwjRJaesX/sponsor/`. HTTP 500 (Internal Server Error) returned, with a HTML body.

## Expected behavior

Return either a HTTP 404 (Not found) or HTTP 400 (Bad Request).

### Relevant log output

```shell
_index=ta_video _id=E8LwjRJaesX found=false

[pid: 40|app: 0|req: 3623/3623] 192.168.1.111 () {44 vars in 701 bytes} [Sat Aug 26 10:51:43 2023] GET /api/video/E8LwjRJaesX/sponsor/ => generated 145 bytes in 27 msecs (HTTP/1.1 500) 7 headers in 248 bytes (1 switches on core 0)
```

**Comment by bbilly1** [comment]
You will find a bunch of those once you start using the API outside of the interface. No need to open bug reports about them, you can go ahead and fix them as you find them. I've already fixed this one for you. :-)

I also added a mention to the docs [here](https://docs.tubearchivist.com/api/introduction/).

**Comment by zac-schutt** [comment]
Awesome! I'm not overly familiar with python, but will fix these as I find them in the same way you've fixed this one.

**Comment by bbilly1** [comment]
that's now published in v0.4.1. If you find any other similar cases, please help fixing them.
